raft/raft.go

```diff
@@ -611,6 +611,7 @@ func (r *raft) becomePreCandidate() {
 	// but doesn't change anything else. In particular it does not increase
 	// r.Term or change r.Vote.
 	r.step = stepCandidate
+	r.votes = make(map[uint64]bool)
```

Comment:
check line612? it says vote should not be changed for some reason. i do not feel it is safe to change the vote decision for one term either. 

/cc @bdarnell